### PR TITLE
add cadl-ranch collectionFormat test

### PR DESCRIPTION
**Pull request by chunyu3**
# Description

- Add cadl-ranch collectionFormat project

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: eng/Generate.ps1 [hunk 1]
```diff
@@ -255,7 +255,8 @@ if (!($Exclude -contains "Samples"))
 $cadlRanchProjectDirectory = Join-Path $repoRoot 'test' 'CadlRanchProjects'
 $cadlRanchProjectPathsWithoutConvenience = # Needs justification to add item
     'enums/extensible', # https://github.com/Azure/autorest.csharp/issues/3079
-    'hello' # https://github.com/Azure/autorest.csharp/issues/3110
+    'hello', # https://github.com/Azure/autorest.csharp/issues/3110
+    'collectionFormat'
```
Comment (pshao25) [inline eng/Generate.ps1:259]: This one has changed to return model, so we could generate convenience method.

File: test/CadlRanchProjects/collectionFormat/Generated/CollectionFormatClient.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace CollectionFormat
+{
+    // Data plane generated client.
+    /// <summary> Test for collectionFormat. </summary>
+    public partial class CollectionFormatClient
```
Comment (pshao25) [inline test/CadlRanchProjects/collectionFormat/Generated/CollectionFormatClient.cs:19]: We could remove this one, right?

Comment (chunyu3) [reply test/CadlRanchProjects/collectionFormat/Generated/CollectionFormatClient.cs:19]: This one is generated from cadl-ranch project collectionFormat. The one we write in the testProjects will contain Header test. 
I will add test for header in cadl-ranch project, then we can remove the one in testProjects

File: eng/Generate.ps1 (hunk 1 above)
Comment (chunyu3) [reply eng/Generate.ps1:259]: We have a defect to generate convenience method https://github.com/Azure/autorest.csharp/issues/3194. We will resolve this first.

File: test/CadlRanchProjects/collectionFormat/Generated/CollectionFormatClient.cs (hunk 2 above)
Comment (chunyu3) [reply test/CadlRanchProjects/collectionFormat/Generated/CollectionFormatClient.cs:19]: File https://github.com/Azure/cadl-ranch/pull/227 to add collection header test.

**Pull request closed without merging**